### PR TITLE
ggplot flowchart

### LINKED ISSUE #74: consider an alternative to `DiagrammeR::grViz()`
@MonikaFu @jacobvjk 

pacta.executive.summary currently uses `DiagrammeR::grViz()` [here](https://github.com/RMI-PACTA/pacta.executive.summary/blob/2909e4b21fc122d58241a21ce66b6b12ffa96c18/R/plot_diagram.R#L33), which creates a htmlwidget, which then requires {chromote} to render in a headless browser to take a screenshot so a static version can be included in the PDF, which requires a Chrome-like browser to be installed on the system. This is a rather significant dependency for the TM Docker image, so we'd like to consider alternative plots/methods for creating [this plot](https://github.com/RMI-PACTA/pacta.executive.summary/blob/2909e4b21fc122d58241a21ce66b6b12ffa96c18/inst/extdata/PA2022CH_en_exec_summary/template.Rmd#L74-L81).

### PR DESCRIPTION
**Pull request by cjyetman**
closes #74 

@MonikaFu for your sanity, this currently assumes an output with a 2:1 width to height ratio

feel free to tweak on top of this PR... setting to draft for now

tested with
``` r
library(pacta.executive.summary)
plot_diagram(toy_data_diagram)
```
Screenshot 2022-11-16 at 14 33 36

### CODE REVIEW

**Comment by AlexAxthelm** [comment]
out of curiosity, why `geom_rect + annotate` rather than `geom_label`?

**Comment by cjyetman** [comment]
> out of curiosity, why `geom_rect + annotate` rather than `geom_label`?

I think what you're asking is why `annotate` versus `geom_label`? The answer to that question is because `annotate` is meant for a single add-on element and does not inherit global settings from the plot, whereas `geom_label` is intended to make multiple elements, one for each row of data, and it does inherit both the data and the global settings from the plot.

Then why `geom_rect`? Because there is no `annotation_rect` and using `annotation_custom` with a rect grob seemed like a pain in the ass.

**Comment by jdhoffa** [comment]
> > out of curiosity, why `geom_rect + annotate` rather than `geom_label`?
> 
> I think what you're asking is why `annotate` versus `geom_label`? The answer to that question is because `annotate` is meant for a single add-on element and does not inherit global settings from the plot, whereas `geom_label` is intended to make multiple elements, one for each row of data, and it does inherit both the data and the global settings from the plot.
> 
> Then why `geom_rect`? Because there is no `annotation_rect` and using `annotation_custom` with a rect grob seemed like a pain in the ass.

NIT: grob is a hilarious word

File: R/plot_diagram.R [hunk 1]
```diff
@@ -30,58 +32,178 @@ plot_diagram <- function(data = NULL) {
...
-    total_exp [label = '{{formatC(data_equity$exposure_portfolio[1], format='f', big.mark=',', digits=0)}} CHF']
-  }
-
-  subgraph cluster_asset_class_exposure {
-    peripheries = 0
+  currency_id <- "CHF"
+  emissions_unit <- "tonnes"
 
-    label = 'Exposure per asset class'
+  ttl_exp_lbl <- paste0(scales::label_comma()(data_equity$exposure_portfolio), " CHF")
```
Comment (jdhoffa) [inline R/plot_diagram.R:38]: Seems like currency should be a parameter?

**Comment by jdhoffa** [comment]
General note, text overlaps if the figure is too small. Maybe this won't ever be a problem if the size of the exec summary is sufficiently wide, but something to be aware of: 

Screen Shot 2022-11-16 at 9 11 48 PM

File: R/plot_diagram.R (hunk 1 above)
Comment (cjyetman) [reply R/plot_diagram.R:38]: yes, but I left decisions like that (like changing the function signature) to @MonikaFu

**Comment by cjyetman** [comment]
> General note, text overlaps if the figure is too small. Maybe this won't ever be a problem if the size of the exec summary is sufficiently wide, but something to be aware of:

Yeah, this plot should have a fixed size, and maybe more importantly a fixed aspect ratio, otherwise things get wonky.

File: R/plot_diagram.R (hunk 1 above)
Comment (cjyetman) [reply R/plot_diagram.R:38]: crap, you're right... I forgot to add the parameter here

Comment (cjyetman) [reply R/plot_diagram.R]: fixed

Comment (jdhoffa) [reply R/plot_diagram.R]: totally fair

**Review by MonikaFu — APPROVED** [review]
I just tweaked the parameters a bit and I would say that this is ready to go!

**Comment by cjyetman** [comment]
should we include re-enabling it in the template here?

**Comment by cjyetman** [comment]
oops, already done

**Comment by jdhoffa** [comment]
Merge ittttttt
https://www.youtube.com/watch?v=ZXsQAXx_ao0

**Comment by jdhoffa** [comment]
ugh, damnit you merged it before I could link the video lol

**Pull request merged**